platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py

```diff
@@ -75,14 +88,47 @@ def get_eeprom_path(self):
         def qsfp_info_get(client):
             return client.pltfm_mgr.pltfm_mgr_qsfp_info_get(self.index)
 
-        if self.get_presence():
-            eeprom_hex = thrift_try(qsfp_info_get)
-            eeprom_raw = bytearray.fromhex(eeprom_hex)
-            with open(self.eeprom_path, 'wb') as fp:
-                fp.write(eeprom_raw)
-            return self.eeprom_path
+        eeprom_hex = thrift_try(qsfp_info_get)
+        eeprom_raw = bytearray.fromhex(eeprom_hex)
+        with open(self.eeprom_path, 'wb') as fp:
+            fp.write(eeprom_raw)
+        return self.eeprom_path
 
-        return None
+    def read_eeprom(self, offset, num_bytes):
+        if not self.get_presence():
+            return None
+
+        if not EEPROM_CACHED_API_SUPPORT:
+            return super().read_eeprom(offset, num_bytes)
+
+
+        # TODO: remove when memory model is verified before requesting page that may not be supported
+        #
+        # If memory model is flat - the requested page can only be 00_LOWER or 00_UPPER,
```

Comment:
This is not correct, can you check [https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L31](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L31)

Reply:
Thank you, removed workaround